src/us/kbase/workspace/database/provenance/CreditMetadata.java

```diff
@@ -0,0 +1,466 @@
+	 * form.
+	 *
+	 * @return resource license (if present)
+	 */
+	public Optional<String> getLicense() {
+		return Optional.ofNullable(license);
+	}
+
+	/*
+	 * Gets the version information for the resource.
+	 *
+	 * @return resource version (if present)
+	 */
+	public Optional<String> getVersion() {
+		return Optional.ofNullable(version);
+	}
+
+	/*
+	 * Gets the resource type.
+	 *
+	 * @return the resource type
+	 */
+	public ResourceType getResourceType() {
+		return resourceType;
+	}
+
+	/*
+	 * Gets the comments for the resource.
+	 *
+	 * @return list of comments (strings)
+	 */
+	public List<String> getComments() {
+		return Common.getList(comments);
+	}
+
+	/*
+	 * Gets the contributor list.
+	 *
+	 * @return list of Contributor objects.
+	 */
+	public List<Contributor> getContributors() {
+		return Common.getList(contributors);
+	}
+
+	/*
+	 * Gets any lifecycle dates.
+	 *
+	 * @return list of EventDate objects
+	 */
+	public List<EventDate> getDates() {
+		return Common.getList(dates);
+	}
+
+	/*
+	 * Gets the funding information.
+	 *
+	 * @return list of FundingReference objects
+	 */
+	public List<FundingReference> getFunding() {
+		return Common.getList(funding);
+	}
+
+	/*
+	 * Gets the list of related identifiers.
+	 *
+	 * @return list of PermanentID objects
+	 */
+	public List<PermanentID> getRelatedIdentifiers() {
+		return Common.getList(relatedIdentifiers);
+	}
+
+	/*
+	 * Gets the list of titles used for the resource.
+	 *
+	 * @return list of Title objects
+	 */
+	public List<Title> getTitles() {
+		return Common.getList(titles);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(comments,
+				contributors,
+				dates,
+				funding,
+				identifier,
+				license,
+				relatedIdentifiers,
+				resourceType,
+				titles,
+				version);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		CreditMetadata other = (CreditMetadata) obj;
+		return Objects.equals(comments, other.comments)
+				&& Objects.equals(contributors, other.contributors)
+				&& Objects.equals(dates, other.dates)
+				&& Objects.equals(funding, other.funding)
+				&& Objects.equals(identifier, other.identifier)
+				&& Objects.equals(license, other.license)
+				&& Objects.equals(relatedIdentifiers, other.relatedIdentifiers)
+				&& Objects.equals(resourceType, other.resourceType)
+				&& Objects.equals(titles, other.titles)
+				&& Objects.equals(version, other.version);
+	}
+
+	/**
+	 * Gets a builder for the {@link CreditMetadata}.
+	 *
+	 * @param identifier
+	 *                unique persistent ID for the resource (i.e. the source data for this
+	 *                workspace object). Should be in the format
+	 *                <database name>:<identifier within database>
+	 * @param resourceType
+	 *                type of the resource, as a {@link ResourceType}
+	 * @param contributors
```

Comment:
This also needs no changes, but inputs are also automatically linked by javadoc so manual links aren't necessary, although they don't hurt
. The link to the CreditMetadata class *is* helpful though.

Reply:
no changes needed :+1: